### PR TITLE
Remove @@species and static this-value support

### PR DESCRIPTION
**Pull request by ptomato**
Closes: #1439

### CODE REVIEW

File: spec/calendar.html
Comment (ljharb) [inline spec/calendar.html:750]: i think we're generally removing the count and enumeration of arguments here (since it's redundant with the method signature)?

**Review by ljharb — CHANGES_REQUESTED** [review]
For ease of review, it'd be really nice if docs, polyfill, and spec changes were in separate commits :-)

File: spec/calendar.html
Comment (ptomato) [reply spec/calendar.html:15]: Forgot to add it to the signature, thanks.

File: spec/calendar.html
Comment (ptomato) [reply spec/calendar.html]: Good! I did find that unnecessary. Is this something that's worth opening a ticket to remove it from the rest of the spec text?

**Comment by ptomato** [comment]
> For ease of review, it'd be really nice if docs, polyfill, and spec changes were in separate commits :-)

Let me know if you need this done for another review round, or if this was OK now that you've looked through it once.

File: spec/calendar.html
Comment (bakkot) [reply spec/calendar.html]: Already done: https://github.com/tc39/ecma262/issues/2304

Comment (ptomato) [reply spec/calendar.html]: Sorry, should've specified, the rest of the Temporal spec text (other than these four methods being changed here). Sounds like yes :smile:

File: spec/calendar.html
Comment (ljharb) [inline spec/calendar.html:15]:
```suggestion
        1. If _newTarget_ is not provided, set it to %Temporal.Calendar%.
```
i think this is the phrasing usually used

**Review by ljharb — COMMENTED** [review]
it would probably help others; i've already manually clicked nearly 300 checkboxes to mark the non-spec files as "viewed" so the hard part's already done for me.

File: spec/plainyearmonth.html
Comment (cjtenny) [inline spec/plainyearmonth.html:73]: Line 69/56 above still defines _constructor_; can be removed

File: spec/plainyearmonth.html
Comment (cjtenny) [inline spec/plainyearmonth.html:550]: Line 571/555 below no longer makes sense / can be removed

File: spec/calendar.html
Comment (cjtenny) [inline spec/calendar.html:633]: I share the confusion re: jordan's prior comment - what is NewTarget here? It seems it must be undefined.

File: spec/calendar.html
Comment (cjtenny) [inline spec/calendar.html:15]: The only place that calls this and provides _newTarget_ is below, where it looks like NewTarget is undefined; if so, then it can be removed from here too.

**Review by cjtenny — APPROVED** [review]
I have shockingly few comments for such a large change. Looks great! Thanks for all the work here, at times my eyes glazed over in the similarity of the test262 bits just reviewing it - I can imagine how frustrating it was to write them all.

Three requested changes before merging, small; two lines to be removed that you missed, and a clarification re: _newTarget_ in spec/calendar which I think can be removed entirely.

File: docs/cookbook/makeExpandedTemporal.mjs
Comment (ptomato) [reply docs/cookbook/makeExpandedTemporal.mjs:82]: It seems like it ought to have been wrong ever since PrepareTemporalFields started coercing the fields to the expected types. But the line `const date = ExpandedTemporal.PlainDate.from({ year: 635427810, month: 2, day: 2 });` still functioned correctly up until the changes made in this PR. Anyway, it's correct now :shrug:

File: spec/calendar.html
Comment (ptomato) [reply spec/calendar.html:633]: Jordan's comment was about a copy-paste error that I believe I fixed — [NewTarget is a spec-text notation that's relevant when evaluating a constructor with the [[Construct]] internal method](https://tc39.es/ecma262/#sec-built-in-function-objects-construct-argumentslist-newtarget) while _newTarget_ is a variable. I had accidentally copy-pasted the latter into this context where I should've used the former, but I believe this is correct now.

Comment (ljharb) [reply spec/calendar.html:633]: that said, it seems kind of nicer to do the NewTarget defaulting in the constructor directly, and let the AO always take a constructor.

Comment (ptomato) [reply spec/calendar.html:633]: The constructor already throws if NewTarget is undefined, so I think this is covered?

File: spec/calendar.html
Comment (bakkot) [inline spec/calendar.html:15]: Nit:

```suggestion
        1. If _newTarget_ is not provided, set _newTarget_ to %Temporal.Calendar%.
```

**Pull request merged**
